### PR TITLE
Fixes #12272 - Support multiple certificates in ovirt resource

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by orrabin** [comment]
@lzap as per our conversation, I changed all the newlines in the example certificates to match the UI and changes the split accordingly.

**Comment by orrabin** [comment]
[test foreman]

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 1]
```diff
@@ -365,9 +365,13 @@ def api_version
       @api_version ||= client.api_version
     end
 
-    def ca_cert_store(cert)
-      return if cert.blank?
-      OpenSSL::X509::Store.new.add_cert(OpenSSL::X509::Certificate.new(cert))
+    def ca_cert_store(certs)
+      return if certs.blank?
+      store = OpenSSL::X509::Store.new
+      certs.split("-----END CERTIFICATE-----").each do |cert|
```
Comment (iNecas) [inline app/models/compute_resources/foreman/model/ovirt.rb]: You can make the split pattern as part of the result like this:

```ruby
certs.split(/(?=-----BEGIN)/)
```

Also, I'm not sure how big chance is the certs would start with `-----BEGIN TRUSTED CERTIFICATE-----` (I saw that format in my pki directory), but keeping the pattern shorter `-----BEGIN` can potentially cover more cases than the specific version that is there right now.

**Comment by orrabin** [comment]
@iNecas thanks, your pattern works better and I don't need to fix the string after it.

**Review by iNecas — APPROVED** [review]
Bonus points for tests :) Waiting for Jenkins

**Comment by iNecas** [comment]
Test failure independent. Merging now. Thanks @orrabin

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 2]
```diff
@@ -365,9 +365,13 @@ def api_version
       @api_version ||= client.api_version
     end
 
-    def ca_cert_store(cert)
-      return if cert.blank?
-      OpenSSL::X509::Store.new.add_cert(OpenSSL::X509::Certificate.new(cert))
+    def ca_cert_store(certs)
+      return if certs.blank?
+      store = OpenSSL::X509::Store.new
+      certs.split(/(?=-----BEGIN)/).each do |cert|
```
Comment (lzap) [inline app/models/compute_resources/foreman/model/ovirt.rb:371]: Nice patch thanks, out of my curiousity, why this is "positive lookahead" regexp?

Comment (iNecas) [reply app/models/compute_resources/foreman/model/ovirt.rb:371]: I not sure I understand the question. It's there to split based on expression, but still include the match inside the split.

Comment (lzap) [reply app/models/compute_resources/foreman/model/ovirt.rb:371]: It's not a match, it's just split into array, interate over it. There is no point to have `?=` (lookahead) option, is my point. Unless @orrabin explains. I am trying to learn here, no big deal.

Comment (orrabin) [reply app/models/compute_resources/foreman/model/ovirt.rb:371]: @lzap my split removed the part of the certificate I was splitting on: `certs.split("-----END CERTIFICATE-----")` and I needed to add it back so @iNecas suggested the lookahead which didn't remove anything.

**Pull request merged**